<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1006</width>
    <height>677</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>911</width>
      <height>601</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>UDP测试</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>886</width>
        <height>556</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_57">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_52">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_53">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_58">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_54">
               <item>
                <widget class="QLabel" name="label_71">
                 <property name="text">
                  <string>本地IP</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_72">
                 <property name="text">
                  <string>本地端口</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_55">
               <item>
                <widget class="QLabel" name="label_73">
                 <property name="text">
                  <string>172.20.10.5</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="lineEdit_31"/>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_59">
             <item>
              <widget class="QPushButton" name="pushButton_59">
               <property name="text">
                <string>开始绑定该端口</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_60">
               <property name="text">
                <string>关闭端口</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_74">
           <property name="styleSheet">
            <string notr="true">color: rgb(0, 170, 255);</string>
           </property>
           <property name="text">
            <string>未绑定端口</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_56">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_60">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_57">
               <item>
                <widget class="QLabel" name="label_75">
                 <property name="text">
                  <string>目标IP</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_76">
                 <property name="text">
                  <string>目标端口</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_77">
                 <property name="text">
                  <string>绑定的组IP</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_58">
               <item>
                <widget class="QLineEdit" name="lineEdit_32"/>
               </item>
               <item>
                <widget class="QLineEdit" name="lineEdit_33">
                 <property name="text">
                  <string>8089</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="lineEdit_34"/>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_61">
             <property name="text">
              <string>绑定组</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_78">
           <property name="text">
            <string>友情提示：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextEdit" name="textEdit_16">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#aa007f;&quot;&gt;多播&lt;/span&gt;IP地址就是D类IP地址,即224.0.0.0至239.255.255.255之间的IP地址。Windows 2000中的DHCP管理器支持多播IP地址的自动分配。&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;在IP网络中，&lt;span style=&quot; color:#aa007f;&quot;&gt;广播&lt;/span&gt;地址用IP地址“255.255.255.255”来表示,这个IP地址代表同一子网内所有。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_59">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_61">
           <item>
            <widget class="QLabel" name="label_79">
             <property name="styleSheet">
              <string notr="true">font: 14pt &quot;Agency FB&quot;;</string>
             </property>
             <property name="text">
              <string>发送区</string>
             </property>
             <property name="textFormat">
              <enum>Qt::AutoText</enum>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_62">
             <item>
              <widget class="QCheckBox" name="checkBox_31">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="text">
                <string>定时发送</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_80">
               <property name="text">
                <string>每隔</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_35">
               <property name="text">
                <string>1000</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_81">
               <property name="text">
                <string>ms</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_82">
           <property name="styleSheet">
            <string notr="true">color: rgb(255, 85, 127);</string>
           </property>
           <property name="text">
            <string>IP广播地址为：172.20.10.5</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_63">
           <item>
            <widget class="QPushButton" name="pushButton_62">
             <property name="text">
              <string>发送</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_63">
             <property name="text">
              <string>广播发送</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBox_32">
             <property name="text">
              <string>按16进制发送</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_64">
             <property name="text">
              <string>清空</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTextEdit" name="textEdit_17">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_64">
           <item>
            <widget class="QLabel" name="label_83">
             <property name="styleSheet">
              <string notr="true">font: 14pt &quot;Agency FB&quot;;</string>
             </property>
             <property name="text">
              <string>接收区</string>
             </property>
             <property name="textFormat">
              <enum>Qt::AutoText</enum>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_65">
             <item>
              <layout class="QGridLayout" name="gridLayout_6">
               <item row="0" column="0">
                <widget class="QCheckBox" name="checkBox_33">
                 <property name="text">
                  <string>按16进制</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QCheckBox" name="checkBox_34">
                 <property name="text">
                  <string>显示时间</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QCheckBox" name="checkBox_35">
                 <property name="text">
                  <string>显示IP地址</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QCheckBox" name="checkBox_36">
                 <property name="text">
                  <string>显示端口</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_65">
               <property name="text">
                <string>清空</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTextEdit" name="textEdit_18">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_60">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_66">
           <item>
            <widget class="QLabel" name="label_84">
             <property name="text">
              <string>ping测试</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_36"/>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_66">
             <property name="text">
              <string>ping</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTextBrowser" name="textBrowser_6"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>TCP服务器</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>TCP客户端</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1006</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
